### PR TITLE
Migrate to new architecture

### LINKED ISSUE #29: Migrate properties panel to new architecture
__What should we do?__

Migrate properties panel to new architecture (similar to https://github.com/bpmn-io/bpmn-js-properties-panel).

- [x] Check if everything is ready for DMN -> no new elements needed
   - [x] camunda/camunda-modeler-design#62
   - [x] camunda/camunda-modeler-design#63
- [x] Prepare for migration - keep only essential files
- [x] Adapt and copy bpmn-js-properties-panel project to fit dmn's needs 
   - Focus only on `drd` for now (even tho we provide properties panel for the decision-table and literal-expression views, we aren't using them)
   - Use [#536](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/536/files)
   - [x] Do migration for `dmn` provider
   - [x] Do migration for `camunda` provider
   - [x] Ensure view changes don't break properties panel
 - [x] Update screenshots
 
New prop panel will be here: https://github.com/bpmn-io/dmn-js-properties-panel/tree/next

__Why should we do it?__

The goal is to have all properties panels used in Camunda Modeler cohesive and conforming with the new properties panel architecture.

### PR DESCRIPTION
**Pull request by smbea**
Closes #29

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -91,7 +91,7 @@ Some things that make it easier for us to accept your pull requests
 * The code is tested
 * The `npm run all` build passes (executes tests + linting)
 * The work is combined into a single commit
-* The commit messages adhere to our [guideline](https://github.com/bpmn-io/bpmn-js/blob/master/docs/project/COMMIT_MESSAGES.md)
+* The commit messages adhere to the [conventional commits guideline](https://www.conventionalcommits.org)
```
Comment (barmac) [inline .github/CONTRIBUTING.md:94]: Good catch!

**Comment by smbea** [comment]
> I've pushed a CI replacement on top of this PR as we don't use Travis anymore. On my machine, `npm run all` fails with
> 
> > Error: No test files found: "test/distro"

I will add these

**Comment by barmac** [comment]
In the old properties panel, there is a short description for the decision ID field:

image

I think if we want to just migrate, this should be added in the Camunda Platform properties provider. For vanilla DMN, there should be no hint.

**Comment by barmac** [comment]
There seems to be a problem with the text annotation:

https://user-images.githubusercontent.com/28307541/152768478-988d78c1-4c1a-4e1e-8b0e-a0a851eac5e7.mov

File: .babelrc [hunk 2]
```diff
@@ -1,5 +1,8 @@
 {
-  "presets": [
-    "env"
+  "plugins": [
+    [ "@babel/plugin-transform-react-jsx", {
+      "importSource": "preact",
```
Comment (barmac) [inline .babelrc:4]: I believe we should use the `@bpmn-io/properties-panel/preact` if we want this to be on par with `bpmn-js-properties-panel`. Cf.  https://github.com/bpmn-io/bpmn-js-properties-panel/blob/master/rollup.config.js#L42

**Comment by philippfromme** [comment]
No icon is displayed when selecting an association. Are we aware of that?

brave_U7YE5IIfYJ

**Comment by smbea** [comment]
> No icon is displayed when selecting an association. Are we aware of that?
> 
> brave_U7YE5IIfYJ

@philippfromme it turns out, that is also the case for bpmn properties panel 🤭

**Comment by smbea** [comment]
@barmac 
I pushed changes for the issues mentioned (except the association icon - see [comment](https://github.com/bpmn-io/dmn-js-properties-panel/pull/31#issuecomment-1031333793)). I rebased the branch to keep the commit history clean so you might have to checkout the branch again.

**Comment by philippfromme** [comment]
Found another issue. Associations shouldn't have a __Name__ entry.

iriAQPdK29

**Comment by smbea** [comment]
I pushed changes which add the Association icon and removed the Name field.

I also created issues for these 2 things for `bpmn-properties-panel`, which I will take on now since they are very simple: https://github.com/bpmn-io/bpmn-js-properties-panel/issues/575, https://github.com/bpmn-io/bpmn-js-properties-panel/issues/576

File: src/provider/camunda/properties/IdProps.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import {
+  useService
+} from '../../../hooks';
+
+import {
+  isIdValid
+} from '../../utils/ValidationUtil';
+
+
+/**
```
Comment (philippfromme) [inline src/provider/camunda/properties/IdProps.js:17]: Indentation should be

```javascript
/**
 *
 */
```

File: src/provider/dmn/properties/NameProps.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+import {
+  useService
+} from '../../../hooks';
+
+import {
+  is
+} from 'dmn-js-shared/lib/util/ModelUtil';
+
+/**
```
Comment (philippfromme) [inline src/provider/dmn/properties/NameProps.js:14]: Indentation broken.

File: test/spec/DmnPropertiesPanel.spec.js [hunk 5]
```diff
@@ -0,0 +1,360 @@
...
+      ];
+    };
+
+    // when
+    const result = createDmnPropertiesPanel({ container, getProviders });
+
+    // then
+    expect(domQueryAll('.bio-properties-panel-group', result.container)).to.have.length(4);
+  });
+
```
Comment (philippfromme) [inline test/spec/DmnPropertiesPanel.spec.js:113]: Add new line. We always put two new lines in between tests.

File: test/spec/PanelHeaderProvider.spec.js [hunk 6]
```diff
@@ -0,0 +1,271 @@
...
+
+      // given
+      const sequenceFlow = createElement('dmn:KnowledgeRequirement');
+
+      // when
+      const type = getConcreteType(sequenceFlow);
+
+      // then
+      expect(type).to.equal('KnowledgeRequirement');
+    });
```
Comment (philippfromme) [inline test/spec/PanelHeaderProvider.spec.js:237]: More new lines please. 😇 

```javascript
    });
    
  });

});
    
```

Comment (smbea) [reply test/spec/PanelHeaderProvider.spec.js:237]: It's all good. Petty makes pretty

Comment (smbea) [reply test/spec/PanelHeaderProvider.spec.js:237]: I pushed changes for this round of feedback (newlines, indentation and revising connections)

**Review by philippfromme — APPROVED** [review]
I've added some minor adjustments regarding code style and the consistent use of getters for business objects (`getBusinessObject(element)`) and moddle properties (e.g. `businessObject.get('id')`).

Great job! Only one properties panel to go! 🚀

File: .babelrc (hunk 2 above)
Comment (barmac) [reply .babelrc:4]:
```suggestion
      "importSource": "@bpmn-io/properties-panel/preact",
```

**Comment by barmac** [comment]
Oops, the bot merged this even though I requested changes @nikku

**Comment by nikku** [comment]
Aaahhhh. :runner:

**Pull request merged**